modules/python-modules/syslogng/modules/s3/s3_destination.py

```diff
@@ -236,14 +239,33 @@ def open(self) -> bool:
         if self.is_opened():
             return True
 
-        self.client = client(
-            service_name="s3",
-            endpoint_url=self.url if self.url != "" else None,
+        self.session = Session(
             aws_access_key_id=self.access_key if self.access_key != "" else None,
             aws_secret_access_key=self.secret_key if self.secret_key != "" else None,
             region_name=self.region,
         )
 
+        if self.role != "":
+            # NOTE: The Session.set_credentials always creates a new Credentials object from the given keys.
+            # NOTE: The DeferredRefreshableCredentials class is a child of RefreshableCredentials which is a
+            # NOTE: child of the Credentials class.
+            self.session._session._credentials = DeferredRefreshableCredentials(
+                refresh_using=create_assume_role_refresher(
+                    self.session.client( "sts" ),
```

Comment:
I see you are creating two clients to the "sts" service. I am not versed in AWS client APIs, but I would guess you could reuse the same client to query "whoami" below, isn't that the case?

Reply:
I think this is fine. This is our open path and it only creates the client side object, probably not even communicating immediately. Approving this.